frontend/src/components/Loading/Loading.tsx

```diff
@@ -56,14 +56,17 @@ export const LoadingPage = React.memo<{ show?: boolean; className?: string }>(
 	({ show, className }) => {
 		const { loadingState } = useAppLoadingContext()
 		const speedFactor = 0.1
+		const shouldShow =
+			show ||
+			[
+				AppLoadingState.LOADING,
+				AppLoadingState.EXTENDED_LOADING,
+			].includes(loadingState)
+		if (!shouldShow) return null
 
 		return (
 			<AnimatePresence>
-				{(show ||
-					[
-						AppLoadingState.LOADING,
-						AppLoadingState.EXTENDED_LOADING,
-					].includes(loadingState)) && (
+				{shouldShow ? (
```

Comment:
Is this needed if we are returning early above?

Reply:
thanks, good catch!